src/Components/Main/Pages/UsersPage/UsersPageContainer.tsx

```diff
@@ -32,23 +37,46 @@ const UsersPageContainer: FC = (): ReactElement | boolean => {
   const followUserCallback = useCallback((id: number) => dispatch(followUser(id)), [dispatch]);
   const unfollowUserCallback = useCallback((id: number) => dispatch(unfollowUser(id)), [dispatch]);
   const getUsersCallback = useCallback(
-    (currentPage: number, currentPageSize: number) => dispatch(getUsers(currentPage, currentPageSize)),
+    (currentPage: number, currentPageSize: number, currentFilter: FilterType, setSubmitting?: SetSubmittingType) =>
+      dispatch(getUsers(currentPage, currentPageSize, currentFilter, setSubmitting)),
     [dispatch]
   );
 
   const mounted: boolean = useMounted();
 
+  const [searchParams, setSearchParams] = useSearchParams();
+
   useEffect(() => {
-    dispatch(getUsers(page, pageSize));
+    const parsedSearchParams = Object.fromEntries(searchParams);
+
+    const actualTerm = parsedSearchParams.term || filter.term;
+    const actualFriend = parsedSearchParams.friend || filter.friend;
+    const actualPage = +parsedSearchParams.page || page;
+
+    const actualFilter = { term: actualTerm, friend: actualFriend };
+
+    dispatch(getUsers(actualPage, pageSize, actualFilter));
     // eslint-disable-next-line
   }, [dispatch]);
 
+  useEffect(() => {
+    const termURL = filter.term === '' ? '' : `&term=${filter.term}`;
+    const friendURL = filter.friend === '' ? '' : `&friend=${filter.friend}`;
+    const pageURL = page === 1 ? '' : `&page=${page}`;
+
+    const queryURL = termURL + friendURL + pageURL;
```

Comment:
For such purpose you can use URLSearchParams =)
https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

Reply:
Done!